src/OpenTelemetry/Metrics/Aggregator.cs

```diff
@@ -0,0 +1,35 @@
+// <copyright file="Aggregator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenTelemetry.Metrics
+{
+    public abstract class Aggregator
+    {
+        public virtual void Update<T>(DateTimeOffset dt, T value)
```

Comment:
virtual generic methods are much slower to call because the runtime has to do a dictionary lookup to figure out what implementation to dispatch to. You'll probably find Aggregator<T> gives you much better perf.

Reply:
Agreed and known. Need to balance the overall design goals while taking advantages of certain language traits.